### PR TITLE
Update Ruby to 3.2 and .github cleanup

### PR DESCRIPTION
**Pull request by kevindew**
This updates this project to the latest version of Ruby (3.2.0) to reflect that Ruby 2.7 will be reaching end-of-life in the coming months.

I did debate whether to go for a lower version of Ruby (3.1.3) since Ruby 3.2.0 is very new but as we don't update this very frequently I felt it was best to go to the latest version. The key advantage of an earlier Ruby version is that devs would likely have that already installed, but Ruby 3.1.3 is also very new so that also doesn't have that advantage and it seemed odd to update this to 3.1.2 when 3.1.3 is out.

### CODE REVIEW

File: spec/govuk_docker/doctor/checkup_spec.rb [hunk 1]
```diff
@@ -20,9 +20,9 @@
   context "when the repository is up-to-date" do
     it "should report that it is up-to-date" do
       subject = described_class.new(
-        service_name: service_name,
+        service_name:,
```
Comment (ChrisBAshton) [inline spec/govuk_docker/doctor/checkup_spec.rb:23]: The [hash value omission feature](https://bugs.ruby-lang.org/issues/14579) seems to have been [added in ruby 3.1.0](https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/), so this commit should probably be `Fix rubocop violations for Ruby >= 3.1`?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -4,5 +4,3 @@ updates:
   directory: "/"
   schedule:
     interval: daily
-    time: "03:00"
-  open-pull-requests-limit: 10
```
Comment (ChrisBAshton) [inline .github/dependabot.yml:8]: Good catch - looks like it was only applied through automation in 448868500934f923f4e93bc935366991e230b7e5.

**Review by ChrisBAshton — APPROVED** [review]
Nice one ⭐

File: spec/govuk_docker/doctor/checkup_spec.rb (hunk 1 above)
Comment (kevindew) [reply spec/govuk_docker/doctor/checkup_spec.rb:23]: Oh yes, good catch. That's what I meant 😆

**Pull request merged**